packages/@aws-cdk/cdk/lib/cloudformation/intrinsics.ts

```diff
@@ -0,0 +1,28 @@
+import { DEFAULT_ENGINE_NAME, ProvisioningEngine, StringFragment } from "../core/engine-strings";
+import { IntrinsicToken  } from "../core/tokens";
+
+/**
+ * The default intrinsics Token engine for CloudFormation
+ */
+export const CLOUDFORMATION_ENGINE = 'cloudformation';
+
+/**
+ * Base class for CloudFormation built-ins
+ */
+export class CloudFormationIntrinsicToken extends IntrinsicToken {
+    protected readonly engine: string = CLOUDFORMATION_ENGINE;
+}
+
+import { FnConcat } from "./fn";
+
+const cloudFormationEngine = {
+    /**
+     * In CloudFormation, we combine strings by wrapping them in FnConcat
+     */
+    combineStringFragments(fragments: StringFragment[]) {
+        return new FnConcat(...fragments.map(f => f.value));
+    }
+};
+
+ProvisioningEngine.register(CLOUDFORMATION_ENGINE, cloudFormationEngine);
+ProvisioningEngine.register(DEFAULT_ENGINE_NAME, cloudFormationEngine);
```

Comment:
We used to at some point because we could at some point stitch values together where none of them would be an intrinsic (and so it would not be obvious which engine to use) but I guess at this point that's something I can do in the framework.